.goreleaser.yaml

```diff
@@ -1,16 +1,98 @@
 # yaml-language-server: $schema=https://goreleaser.com/static/schema.json
 # vim: set ts=2 sw=2 tw=0 fo=cnqoj
+env:
+  - CGO_ENABLED=1
 
 builds:
-  - id: autokitteh-starlark-lsp
+  - id: darwin-arm64
     binary: autokitteh-starlark-lsp
     main: ./cmd/autokitteh-starlark-lsp
     goos:
-      - linux
       - darwin
     goarch:
-      - amd64
       - arm64
+    env:
+      - CC=o64-clang
+      - CXX=o64-clang++
+      - PKG_CONFIG_SYSROOT_DIR=/sysroot/macos/arm64
+      - PKG_CONFIG_PATH=/sysroot/macos/arm64/usr/local/lib/pkgconfig
+    flags:
+      - -mod=readonly
+    ldflags:
+      - -s -w -X main.version={{.Version}}
+
+  - id: darwin-amd64
+    binary: autokitteh-starlark-lsp
+    main: ./cmd/autokitteh-starlark-lsp
+    goos:
+      - darwin
+    goarch:
+      - amd64
+    env:
+      - CC=o64-clang
+      - CXX=o64-clang++
+      - PKG_CONFIG_SYSROOT_DIR=/sysroot/macos/amd64
+      - PKG_CONFIG_PATH=/sysroot/macos/amd64/usr/local/lib/pkgconfig
+    flags:
+      - -mod=readonly
+    ldflags:
+      - -s -w -X main.version={{.Version}}
+
+  - id: linux-amd64
+    binary: autokitteh-starlark-lsp
+    main: ./cmd/autokitteh-starlark-lsp
+    goos:
+      - linux
+    goarch:
+      - amd64
+    env:
+    flags:
+      - -mod=readonly
+    ldflags:
+      - -s -w -X main.version={{.Version}}
+
+  - id: linux-armhf
+    binary: autokitteh-starlark-lsp
+    main: ./cmd/autokitteh-starlark-lsp
+    goos:
+      - linux
+    goarch:
+      - arm
+    goarm:
+      - 7
+    env:
+      - CC=arm-linux-gnueabihf-gcc
+      - CXX=arm-linux-gnueabihf-g++
+      - CGO_CFLAGS=--sysroot=/sysroot/linux/armhf
+      - CGO_LDFLAGS=--sysroot=/sysroot/linux/armhf
+      - PKG_CONFIG_SYSROOT_DIR=/sysroot/linux/armhf
+      - PKG_CONFIG_PATH=/sysroot/linux/armhf/opt/vc/lib/pkgconfig:/sysroot/linux/armhf/usr/lib/arm-linux-gnueabihf/pkgconfig:/sysroot/linux/armhf/usr/lib/pkgconfig:/sysroot/linux/armhf/usr/local/lib/pkgconfig
```

Comment:
this assumes you run this on a specific machine with specific arch ?
maybe a comment will be helpful here

Reply:
no, it's a cross compiling. docker is the same docker. But there are several compilers and related things installed. 
The underlying arch of the docker is linux amd64, unrelated to target arch